### ISSUE TITLE
The overlay changes too abruptly

### ISSUE THREAD

**Issue by monolithed** [Farzin-Firoozi/react-modern-drawer#54]
The overlay changes too abruptly with a flash. It’s likely necessary to animate the `background-color` the way it's done in Ant Design and other libraries.

https://github.com/user-attachments/assets/0e60c27f-06ab-41e1-910e-cae4ba2cf384

**Comment by AdrianMrn** [comment]
Agreed, would be nice to have the overlay animate from opacity 0 to 0.4 (or whichever value you have set for your final opacity). This can easily be done with css, @Farzin-Firoozi if you want I can make a PR?

**Comment by Farzin-Firoozi** [comment]
Thank you for the suggestion! Yes, please go ahead and make the PR with the CSS animation for the overlay. Would be great to see it in action!

**Comment by namti** [comment]
Hi @Farzin-Firoozi , I have fixed this and sent a pull requests. Please review. Thanks
